internal/dcache/replication_manager/utils.go

```diff
@@ -165,3 +161,25 @@ func getCachePathForRVName(rvName string) string {
 
 	return rv.LocalCachePath
 }
+
+// Update the state of the RV in the given component RVs list.
+func updateLocalComponentRVState(rvs []*models.RVNameAndState, rvName string,
+	oldState dcache.StateEnum, newState dcache.StateEnum) {
+	common.Assert(len(rvs) == int(getNumReplicas()), len(rvs), getNumReplicas())
+	common.Assert(cm.IsValidRVName(rvName), rvName)
+	common.Assert(oldState != newState &&
+		cm.IsValidComponentRVState(oldState) &&
+		cm.IsValidComponentRVState(newState), rvName, oldState, newState)
+
+	for _, rv := range rvs {
+		common.Assert(rv != nil, "Component RV is nil")
+		if rv.Name == rvName {
+			common.Assert(rv.State == string(oldState), rvName, rv.State, oldState)
+			rv.State = string(newState)
+			break
```

Comment:
you meant return?
did you not test this code? o/w the following assert should have failed.